presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/ExpressionConverter.java

```diff
@@ -168,14 +162,6 @@ else if (highBound == BELOW && high.getValueBlock().isPresent()) {
 
     private static Object getValue(Type type, Marker marker, ConnectorSession session)
     {
-        if (type instanceof TimestampWithTimeZoneType || type instanceof TimeWithTimeZoneType) {
-            return MILLISECONDS.toMicros(unpackMillisUtc((Long) marker.getValue()));
-        }
-
-        if (type instanceof TimestampType || type instanceof TimeType) {
```

Comment:
I just realized that this code is actually correct. The problem is that we aren't doing any conversion between millis and micros when reading and writing the data files. With the removal of this conversion code, we are now pushing a millis value into the Iceberg predicate code, which works because the files are written with a millis value.

So it works from a round trip perspective, but it's not valid from an Iceberg format perspective. This is where having integration tests against Spark, or read/write directly with the Iceberg library, would be helpful.

Reply:
Nevermind, as @lxynov pointed out, ORC supports timestamps natively, so we're reading and writing them correctly.